app/services/service_creator.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class ServiceCreator
+
+  delegate :backend_api_configs, :backend_api_proxy, :account, to: :@service
+  delegate :provider_can_use?, to: :account
+
+  def initialize(service:, backend_api: nil)
+    @service     = service
+    @backend_api = backend_api
+  end
+
+  def call!(params = {})
+    @service.transaction do
+      backend_api_proxy_params = params.dup
+      service_params = backend_api_proxy_params.slice!(:path, :private_endpoint)
+      @service.attributes = service_params
+      save!(backend_api_proxy_params)
+    end
+  end
+
+  def call(params = {})
+    call!(params)
+  rescue ActiveRecord::RecordInvalid
+    false
+  end
+
+  private
+
+  def save!(params)
+    @service.save!
+    if @backend_api
+      save_assigned_backend_api(params)
+    else
+      save_default_backend_api(params)
+    end
+  end
+
+  def save_default_backend_api(params)
```

Comment:
```suggestion
  def save_default_backend_api!(params)
```

...as it ends with `update!`.

Despite Code Climate complaining, I don't really about having a quiet version without the bang.

Reply:
Hmm. I see now they are `private`. Maybe not needed then.